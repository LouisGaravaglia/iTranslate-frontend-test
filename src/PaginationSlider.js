import React, {useState, memo} from 'react';
import {withStyles} from '@material-ui/core/styles';
import Slider from '@material-ui/core/Slider';

//CUSTOM STYLING FOR MUI SLIDER
const MyPaginationSlider = withStyles({
    root: {
    color: "#fff",
    height: 8
  },
  thumb: {
    height: 24,
    width: 24,
    backgroundColor: "#fff",
    border: "2px solid currentColor",
    marginTop: -10,
    marginLeft: -12,
    "&:focus,&:hover,&$active": {
      boxShadow: "inherit"
    }
  },
  active: {},
  valueLabel: {
    left: "calc(-50% + 14px)",
    top: -22,
    "& *": {
      background: "transparent",
      color: "#000"
    }
  },
  track: {
    height: 8,
    borderRadius: 4
  },
  rail: {
    height: 4,
    borderRadius: 4,
    opacity: 0.5,
    backgroundColor: "#bfbfbf"
  }
})(Slider);

const PaginationSlider = memo(({resultsArray, itemsPerPage, handleSliderChange, containerClass, sliderClass}) => {
  const [sliderVal, setSliderVal] = useState(0);
  const multipleOf = resultsArray.length % itemsPerPage === 0;
  let maxSliderVal = Math.floor(resultsArray.length / itemsPerPage);

  if (!multipleOf) maxSliderVal += 1;

  const handleChange = (event, newValue) => {
    setSliderVal(newValue);
    handleSliderChange(newValue);
  };

////////////////////////////////////////////////////  RETURN  ////////////////////////////////////////////////////

  return (
    <div className={containerClass}>
      <MyPaginationSlider className={sliderClass} color="" value={sliderVal} max={maxSliderVal - 1} min={0} step={1} onChange={handleChange} aria-labelledby="continuous-slider" track={false}/>
    </div>
  );
});

export default PaginationSlider;