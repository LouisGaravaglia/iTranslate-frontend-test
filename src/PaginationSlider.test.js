import React from 'react'
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import {Provider} from 'react-redux'
import {createStore} from "redux";
import PaginationSlider from "./PaginationSlider";

afterEach(cleanup);

const startingState = {};

/**
 * reducer function to mimic the reducer used in Redux.
 * @param {object} state - holds an object of data used within the components
 * @param {object} action - object where the values are strings used to fire certain actions
 */
function reducer(state = startingState, action) {
  switch (action.type) {
    case "ADD_TRACKS":
      return {...state, tracks: action.tracks};
    default:
      return state;
  };
};

/**
 * Function that renders the component within a redux environment since most 
 * components rely on redux for data in order to mount.
 * @param {ReactComponent} component - private spotify client id
 * @param {function} store - creates the react store using the createStore method imported from redux
 */
function renderWithRedux(
  component,
  {initialState, store = createStore(reducer, initialState)} = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>)
  };
};

//SMOKE TEST
describe('Smoke Test for PaginationSlider component', () => {

  it('renders without crashing', () => {
    renderWithRedux(<PaginationSlider resultsArray={[{trackId: 1}, {trackId: 2}]}/>);
  });
});

//SNAPSHOT TEST
describe('Snapshot Test for PaginationSlider component', () => {

  it('matches snapshot', () => {
    const {asFragment} = renderWithRedux(<PaginationSlider resultsArray={[{trackId: 1}, {trackId: 2}]}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});