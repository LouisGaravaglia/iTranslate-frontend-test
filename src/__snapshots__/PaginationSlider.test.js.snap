// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for PaginationSlider component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span
      class="MuiSlider-root WithStyles(ForwardRef(Slider))-root-12 MuiSlider-trackFalse"
    >
      <span
        class="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-17"
      />
      <span
        class="MuiSlider-track WithStyles(ForwardRef(Slider))-track-16"
      />
      <input
        type="hidden"
        value="NaN"
      />
      <span
        aria-labelledby="continuous-slider"
        aria-orientation="horizontal"
        aria-valuemax="NaN"
        aria-valuemin="0"
        aria-valuenow="NaN"
        class="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-13"
        data-index="0"
        role="slider"
        tabindex="0"
      />
    </span>
  </div>
</DocumentFragment>
`;
